iocage/lib/Network.py

```diff
@@ -21,75 +21,106 @@
 # STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING
 # IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
 # POSSIBILITY OF SUCH DAMAGE.
+import typing
 import subprocess  # nosec: B404
 from hashlib import sha224
 
+import iocage.lib.BridgeInterface
 import iocage.lib.NetworkInterface
 import iocage.lib.errors
 import iocage.lib.helpers
 
 
 class Network:
-    def __init__(self, jail,
-                 nic="vnet0",
-                 ipv4_addresses=None,
-                 ipv6_addresses=None,
-                 mtu=1500,
-                 bridges=None,
-                 logger=None):
 
-        self.logger = iocage.lib.helpers.init_logger(self, logger)
+    def __init__(
+        self,
+        jail: 'iocage.lib.Jail.JailGenerator',
+        nic: typing.Optional[str]="vnet0",
+        ipv4_addresses: typing.Optional[typing.List[str]]=None,
+        ipv6_addresses: typing.Optional[typing.List[str]]=None,
+        mtu: typing.Optional[int]=1500,
+        bridge: typing.Optional[
+            'iocage.lib.BridgeInterface.BridgeInterface'
+        ]=None,
+        logger: 'iocage.lib.Logger.Logger'=None
+    ) -> None:
 
-        if bridges is not None:
-            if not isinstance(bridges, list):
-                raise iocage.lib.errors.InvalidNetworkBridge(
-                    reason="Expected None or list of strings",
-                    logger=self.logger
-                )
+        self.logger = iocage.lib.helpers.init_logger(self, logger)
 
         self.vnet = True
-        self.bridges = bridges
+        self.bridge = bridge
         self.jail = jail
         self.nic = nic
         self.mtu = mtu
         self.ipv4_addresses = ipv4_addresses or []
         self.ipv6_addresses = ipv6_addresses or []
 
-    def setup(self):
-        if self.vnet:
-            if not self.bridges or len(self.bridges) == 0:
-                raise iocage.lib.errors.VnetBridgeMissing(
-                    logger=self.logger
-                )
+    def setup(self) -> None:
+        if (self.vnet is True) and (self.bridge is None):
+            raise iocage.lib.errors.VnetBridgeMissing(logger=self.logger)
+        self.__create_vnet_iface()
+
+    def teardown(self) -> None:
+        if self.vnet is True:
+            self.__down_host_interface()
+            if self.bridge.secure is True:
+                self.__down_secure_mode_devices()
+
+    def __down_host_interface(self) -> None:
```

Comment:
regardless if it is or isn't, this name is confusing.

Reply:
do we have a standard in our code for how to name functions?

`__verb_object_subject(object, ...)`?
`__verb_subject_object(subject, ...)`?
`__verb_object_object(subject, …)?`

is this all confusing enough?

Reply:
No standard. It's a private function - it should be easy to read in this files context.

Interesting idea to standardize naming of transformation functions like this. We should make them idempotent too 🤓